### ISSUE TITLE
Find and Develop another data storage

### ISSUE THREAD

**Issue by soyto** [soyto/soyto.github.io#9]
github data storage limit is about [1GB](https://help.github.com/articles/what-is-my-disk-quota/) (Thanks to [Ayone from board.en forum](http://board.en.aion.gameforge.com/board6-aion-the-community/board56-aion-fanbase/board135-user-projects/p268391-soyto-ranking-tool/#post268391)) 

I just thinked about use some free mongoDB storage (Data isn't as big) and look for a free nodeJS server to create a full RESTFULL JSONP acces to this mongoDB database.

The architecture that the site will have must be the follow one:

- gihub.io as front page with AngularJS
- NodeJS with RESTFULL services availables with JSONP
- MongoDB storage

What do you think about this architecture?

**Comment by iltar** [comment]
I'm not familiar with NodeJS development, but have you taken a look at [React](https://facebook.github.io/react/) or [React PHP](https://github.com/reactphp/react)? Did you have in mind what kind of back-end you want, PHP, JS, any Framework? 

_Anyone_ :+1:

**Comment by soyto** [comment]
I will explain why i want to make 3 tiers application.

- For data storage i am now using large JSON files with complex structure, app is based on read features and have no write atm. That's why i want to chose MongoDB up to another relational Database.
- MondoDB provides RESTFULL access but we want to use as 1st tier app github io  (read next point). That's why we need some mid tier that can handle JSONP and also add some security over our MongoDB.
- github.io is fast serving html and js files, that's better than another free hosting. We want to serve html files with angularJS, and with angularJS we will access to our 2nd tier (Application tier) for retrieving data

**Comment by soyto** [comment]
After some tests and some parallel development i reach the conclusion that there is no free hosting that will meet system requirements.

First i tried Openshift (https://openshift.redhat.com/) and i reached fast storage quota problems when migrating data. (1GB of storage for DB and app isn't enough)

Then i tried Mongolab (https://mongolab.com/) and readhed so fast the storage limits (500MB).


That's what i am going to make now:

- Cutdown some data, just 1 month will be shown in application.
- I will use OneDrive folder for store whole server data (I will give you away a link to this folder).
- Add a donate button, if people are good enough to give me some money i will continue the nodeJS and MongoDB development

**Comment by iltar** [comment]
My company sells VPS with 25GB of storage and 1TB of traffic, 10 Euro/month, [upgrades are available]( https://www.hostnet.nl/virtual-private-servers/vps-prijzen-en-bestellen). It's in Dutch, but something like this might a viable solution.

**Comment by soyto** [comment]
Yeah that's right, will be a great solution. But atm i am not going to buy any host provider or virtual machine for this project. 

In a future, if people start to donate some money i will look to buy somone.

**Comment by soyto** [comment]
Bad comment on commit
